apps/meteor/server/models/raw/ModerationReports.ts

```diff
@@ -132,6 +132,83 @@ export class ModerationReportsRaw extends BaseRaw<IModerationReport> implements
 		return this.col.aggregate(params, { allowDiskUse: true });
 	}
 
+	findUserReports(
+		latest: Date,
+		oldest: Date,
+		selector: string,
+		pagination: PaginationParams<IModerationReport>,
+	): AggregationCursor<Pick<UserReport, '_id' | 'reportedUser' | 'ts'> & { count: number }> {
+		const query = {
+			_hidden: {
+				$ne: true,
+			},
+			ts: {
+				$lt: latest,
+				$gt: oldest,
+			},
+			message: {
+				$exists: false,
+			},
+			...this.getSearchQueryForSelectorUsers(selector),
+		};
+
+		const { sort, offset, count } = pagination;
+
+		const pipeline = [
+			{ $match: query },
+			{
+				$sort: {
+					ts: -1,
+				},
+			},
+			{
+				$group: {
+					_id: '$reportedUser._id',
+					count: { $sum: 1 },
+					reports: { $first: '$$ROOT' },
+				},
+			},
+			{
+				$sort: sort || {
+					'reports.ts': -1,
+				},
+			},
+			{
+				$skip: offset,
+			},
+			{
+				$limit: count,
+			},
+			{
+				$project: {
+					_id: 0,
+					reportedUser: '$reports.reportedUser',
+					ts: '$reports.ts',
+					count: 1,
+				},
+			},
+		];
+
+		return this.col.aggregate(pipeline, { allowDiskUse: true, readPreference: readSecondaryPreferred() });
+	}
+
+	async getTotalUniqueReportedUsers(latest: Date, oldest: Date, selector: string, isMessageReports?: boolean): Promise<number> {
+		const query = {
+			_hidden: {
+				$ne: true,
+			},
+			ts: {
+				$lt: latest,
+				$gt: oldest,
+			},
+			...(isMessageReports ? this.getSearchQueryForSelector(selector) : this.getSearchQueryForSelectorUsers(selector)),
+		};
+
+		const field = isMessageReports ? 'message.u._id' : 'reportedUser._id';
+		const ids = await this.col.distinct(field, query);
```

Comment:
you don't need to fetch **all data** to know how many different values are.. you can can use an aggregation for that, something like:
```js
{
  $group: {
    _id: '$message.u._id', // group all values into a single _id key
  }
},
{
  $group: {
    _id: 'total', // group all previously values into a single value
    total: { $sum: 1 }, // sum the total values from previous pipeline
  }
}
```

Reply:
Thank you, yes, we could do it, but I got to learn in the past, we had some issues with aggregation, so we should try to avoid the `aggregation` as much as possible. Please let me know if I misunderstood something.

Reply:
Had a discussion on this, and made the suggested changes. Thank you!